gapis/api/templates/specific_gfx_api.cpp.tmpl

```diff
@@ -145,7 +145,13 @@
         {{$args := (GetAnnotation $ "indirect").Arguments}}
         {{$elem := (index $.CallParameters 0).Name}}
         {{$func := Macro "CmdName" $}}
-        if ({{Template "GetIndirectedFunction" "Annotations" $args "Element" $elem "Function" $func}}) {
+        if (
+        #ifdef DEBUG
+        {{Template "GetIndirectedFunction" "Annotations" $args "Element" $elem "Function" $func}}
+        #else
+        true
```

Comment:
Should this be `false`, to make the `if` body dead-code when not in DEBUG mode?

Reply:
We still want to call the API function, we just don't want to check all the pointers and handles for validity before doing it.

Reply:
OK, thanks for the clarification.

Reply:
Having a preprocessor macro inside of a condition is ... awkward.

```
if (
#ifdef
.....
#else
true
#endif
) {
```
I would prefer we generate something slightly nicer.

```
#ifndef NDEBUG
if (!(...)) {
   GAPID_ERROR("Error during calling function {{$name}}");
      return false;
}
#endif
.....
```

Reply:
If we did that we'd call ATrace_beginSection() without calling ATrace_endSection() and any other post-call logic would need to be duplicated. All of the API file code is "awkward" :)